components/hab/src/command/origin/secret/generate.rs

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) 2016-2018 Chef Software Inc. and/or applicable contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use depot_client::Client as DepotClient;
+use common::ui::{Status, UI};
+use super::super::key::download::download_public_encryption_key;
+
+use hcore::crypto::BoxKeyPair;
+use {PRODUCT, VERSION};
+use error::{Error, Result};
+use std::path::Path;
+
+pub fn start(
+    ui: &mut UI,
+    bldr_url: &str,
+    token: &str,
+    origin: &str,
+    key: &str,
+    secret: &str,
+    cache: &Path,
+) -> Result<()> {
+    let depot_client = DepotClient::new(bldr_url, PRODUCT, VERSION, None).map_err(
+        Error::DepotClient,
+    )?;
+
+    let encryption_key = match BoxKeyPair::get_latest_pair_for(origin, cache) {
+        Ok(key) => key,
+        Err(_) => {
+            debug!("Didn't find public encryption key in cache path");
+            download_public_encryption_key(ui, &depot_client, origin, token, cache)?;
+            BoxKeyPair::get_latest_pair_for(origin, cache)?
+        }
+    };
+
+    ui.status(Status::Encrypting, format!("value for key {}.", key))?;
+    let encrypted_secret_bytes = encryption_key.encrypt(secret.as_bytes(), None)?;
+    let encrypted_secret_string = String::from_utf8_lossy(&encrypted_secret_bytes);
```

Comment:
If this is arbitrary bytes, I worry about storing them in a `str`, which guarantees utf8 validity. Can we switch to either base64 encoding or using a `Vec<u8>` in `create_origin_secret`?

Reply:
I think I can switch `from_utf8_lossy` to `from_utf8` to get a utf8 guarantee? What do you think?

Reply:
Done :tada:

Reply:
I think that you're going to get errors at the rate which a random series of bytes do not make valid UTF-8, right? I don't know what that frequency is off the top of my head, but if there a reason why you wouldn't?

Reply:
I'm not sure I follow the random series of bytes bit. We are taking a user supplied string, converting it to bytes, sending it to encrypt which returns a base64 encoded series of bytes, then we convert that back into a string for a json payload. Does that match up with what you were thinking?

Reply:
For posterity, the confusion was mine. I did not realize that `BoxKeyPair::encrypt` returned already base64-encoded data. I assumed since it's return type was `Result<Vec<u8>>` that it was raw bytes. This is pretty misleading, so I filed an issue: https://github.com/habitat-sh/core/issues/15